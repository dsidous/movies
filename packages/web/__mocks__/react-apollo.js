const apollo = require.requireActual('react-apollo');
const result = jest.fn(Component => Component);

apollo.graphql = jest.fn().mockReturnValue(result);

module.exports = apollo;
