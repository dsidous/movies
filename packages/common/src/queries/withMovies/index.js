import { graphql } from 'react-apollo';
import { query } from './query';

const withMovies = () =>
  graphql(query, {
    options: props => ({
      fetchPolicy: 'cache-first',
      variables: {
        query: props.query,
      },
    }),
    props: ({ data: { movies = [], loading } }) => ({
      loading,
      shows: movies,
    }),
  });

export default withMovies;
