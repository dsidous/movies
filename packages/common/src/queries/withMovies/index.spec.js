import { graphql } from 'react-apollo';
import React from 'react';
import { shallow } from 'enzyme';
import withMovies from '.';

jest.mock('react-apollo', () => ({
  ...jest.requireActual('react-apollo'),
  graphql: jest.fn(),
}));

describe('Queries/withMovies', () => {
  graphql.mockImplementation((_, config) => {
    const ownProps = { query: 'page=1' };
    const props = config.props({
      data: {
        movies: [
          {
            id: 1,
            backdrop_path: 'path/to/image',
            poster_path: 'path/to/image',
            title: 'title',
            overview: 'overview',
            release_date: '01.01.2010',
            vote_average: 1,
            genre_ids: [1],
          },
        ],
        loading: true,
      },
      ownProps,
    });
    const options = config.options(ownProps);
    return jest.fn(Mock => <Mock options={options} {...props} />);
  });

  const MockComponent = props => ({ ...props });
  const Component = withMovies()(MockComponent);

  it('should render as expected', () => {
    const render = shallow(Component);
    expect(render).toMatchSnapshot();
  });
});
