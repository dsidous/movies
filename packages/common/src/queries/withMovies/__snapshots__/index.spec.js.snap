// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries/withMovies should render as expected 1`] = `ShallowWrapper {}`;
